lib/product.rb

```diff
@@ -0,0 +1,22 @@
+class Product
+  attr_reader :category, :name, :unit_price, :quantity, :hoard
+  def initialize(category, name, unit_price, quantity)
+    @category = category
+    @name = name
+    @unit_price = unit_price
+    @quantity = quantity.to_i
+    @total_price = total_price
+
```

Comment:
We should have an instance variable for `@is_hoarded` here that defaults to `false` - that would help with the is_hoarded? method below.